### PR TITLE
feat(benchmark): added an implementation of the tree benchmark in React

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

**Comment by tbosch** [comment]
Benchmarks from my computer:

```
React create tree:
- script time: 82.07+-3%
- gc amount: 13267.07

React update tree: 32.19+-6%

Tree create (viewcache):
- script time: 42.56+-2%
- gc amount: 3518.74+-173%

Tree create (no viewcache):
- script time: 97.66+-7%
- gc amount: 13495.11+-29%

Tree update: 7.13+-2%

Baseline create: 20.29+-3%
Baseline update: 1.27+-8%
```

File: modules/angular2/src/core/compiler/view_manager_utils.js [hunk 1]
```diff
@@ -208,6 +207,7 @@ export class AppViewManagerUtils {
       }
     }
 
+    view.changeDetector.hydrate(view.context, view.locals, view);
```
Comment (tbosch) [inline modules/angular2/src/core/compiler/view_manager_utils.js]: Could you add a unit test for this?

**Comment by tbosch** [comment]
Measured again with `JitChangeDetection`:

```
ng2 create (no cache): 92.99+-12%
ng2 create (viewcache): 38.54+-5%
ng2 update: 5.18+-5%

baseline create: 20.14+-4%
baseline update: 1.26+-7%

react create: 84.44+-9%
react update: 34.20+-9%
```

File: package.json [hunk 2]
```diff
@@ -97,6 +97,7 @@
     "protractor": "2.0.0",
     "q": "^1.0.1",
     "qq": "^0.3.5",
+    "react": "^0.13.2",
```
Comment (IgorMinar) [inline package.json]: you haven't added react to our shrink-wrap, so the install will not happen on ci or for other developers. please read `npm-shrinkwrap.readme.md`.

File: protractor-dart2js.conf.js [hunk 3]
```diff
@@ -7,7 +7,8 @@ config.exclude.push(
   'dist/js/cjs/examples/e2e_test/sourcemap/sourcemap_spec.js',
   // TODO: remove this line when largetable dart has been added
   'dist/js/cjs/benchmarks_external/e2e_test/largetable_perf.js',
-  'dist/js/cjs/benchmarks_external/e2e_test/polymer_tree_perf.js'
+  'dist/js/cjs/benchmarks_external/e2e_test/polymer_tree_perf.js',
+  'dist/js/cjs/benchmarks_external/src/tree/react/index.es6'
```
Comment (tbosch) [inline protractor-dart2js.conf.js]: Path should be `dist/js/cjs/benchmarks_external/e2e_test/tree/react/index.js`

File: protractor-dart2js.conf.js [hunk 4]
```diff
@@ -7,7 +7,8 @@ config.exclude.push(
   'dist/js/cjs/examples/e2e_test/sourcemap/sourcemap_spec.js',
   // TODO: remove this line when largetable dart has been added
   'dist/js/cjs/benchmarks_external/e2e_test/largetable_perf.js',
-  'dist/js/cjs/benchmarks_external/e2e_test/polymer_tree_perf.js'
+  'dist/js/cjs/benchmarks_external/e2e_test/polymer_tree_perf.js',
+  'dist/js/cjs/benchmarks_external/e2e_test/tree/react/index.js'
```
Comment (tbosch) [inline protractor-dart2js.conf.js]: ...`benchmarks_external/e2e_test/react_tree_perf.js`

**Pull request merged**